src/snapred/ui/presenter/TestPanelPresenter.py

```diff
@@ -1,27 +1,98 @@
     worker_pool = WorkerPool()
 
     def __init__(self, view):
+        reductionRequest = SNAPRequest(path="api", payload=None)
+        self.apiDict = self.interfaceController.executeRequest(reductionRequest).responseData
+
+        self.apiComboBox = self.setupApiComboBox(self.apiDict, view)
+
+        jsonSchema = json.loads(self.apiDict["config"][""]["runs"])
         self.view = view
+        self.jsonForm = JsonForm("Advanced Parameters", jsonSchema=jsonSchema, parent=view)
+        self._loadDefaultJsonInput("config//runs", self.jsonForm)
+        self.comboSelectionView = BackendRequestView(self.jsonForm, "config//runs", parent=self.view)
+        self.view.centralWidget.layout().addWidget(self.apiComboBox)
+        self.view.centralWidget.layout().addWidget(self.comboSelectionView)
+        self.view.centralWidget.layout().setAlignment(self.comboSelectionView, Qt.AlignTop | Qt.AlignHCenter)
+        self.view.adjustSize()
+
+    def _getPaths(self, apiDict):
+        paths = []
+        for key, value in apiDict.items():
+            if type(value) is dict:
+                subpaths = self._getPaths(value)
+                paths.extend(["{}/{}".format(key, path) for path in subpaths])
+            else:
+                paths.append(key)
+        return paths
+
+    def setupApiComboBox(self, apiDict, parent=None):
+        comboBox = QComboBox(parent)
+        for path in self._getPaths(apiDict):
+            comboBox.addItem(path)
+
+        comboBox.currentIndexChanged.connect(self.handleApiComboSelected)
+        return comboBox
+
+    def _findSchemaForPath(self, path):
+        currentVal = self.apiDict
+        # TODO: Replace with Config
+        subPaths = path.split("/")
+        for subpath in subPaths:
+            currentVal = currentVal[subpath]
+        return currentVal
+
+    def _getSchemaForSelection(self, selection):
+        schemaString = self._findSchemaForPath(selection)
+        return json.loads(schemaString) if schemaString else {}
+
+    def _loadDefaultJsonInput(self, selection, jsonForm):
+        subPaths = selection.split("/")
+        subPaths.pop(-1)
+        if subPaths[-1] == "":
+            subPaths.pop(-1)
+        defaultFilePath = "default/request/" + "/".join(subPaths) + "/payload.json"
+        if Resource.exists(defaultFilePath):
+            defaults = json.loads(Resource.read(defaultFilePath))
+            jsonForm.updateData(defaults)
+        else:
+            logger.warn("No default values for path: {}".format(defaultFilePath))
+
+    def handleApiComboSelected(self, index):  # noqa: ARG002
+        selection = self.apiComboBox.currentText()
+        jsonSchema = self._getSchemaForSelection(selection)
+        # import pdb;pdb.set_trace()
+        newForm = JsonForm(selection.split("/")[-1], jsonSchema=jsonSchema, parent=self.view)
+        self._loadDefaultJsonInput(selection, newForm)
+        if selection.startswith("calibration/reduction"):
```

Comment:
Does python not have switch structures?  This method of branching on the button name seems suboptimal.

Reply:
like a map of path, view pairs that defaults to the default view if the path doesnt have a custom screen in the map.